### ISSUE TITLE
gcn-link-prediction 0% accuracy 

### ISSUE THREAD

**Issue by amrlnic** [stellargraph/stellargraph#2033]
### Describe the bug

*In the notebook gcn-link-prediction.ipynb model training remains stuck at 0% accuracy.*
Link: https://github.com/stellargraph/stellargraph/blob/9370caed1137c2527523a39212072df1760ca00f/docs/demos/link-prediction/gcn-link-prediction.nblink

### To Reproduce

1. *Go to [this notebook](https://colab.research.google.com/github/stellargraph/stellargraph/blob/master/demos/link-prediction/gcn-link-prediction.ipynb)*
2. *Execute 'Runtime -> Run all'*
3. *See error (Evaluation output at the end)*

### Observed behavior

*In the output during training or in the evaluation output at the end we can see how the accuracy is 0%*

### Expected behavior

*An improvement in accuracy was expected during training (even just a variation of the values).*

### Environment

Operating system: *Colab*

Python version: *3.7.12*

Package versions: *stellargraph==1.2.1*

### Additional context

Add any other context about the problem here. For example:

Data used: *CORA network data (Notebook default dataset)*

Screenshots: 

![image](https://user-images.githubusercontent.com/43885625/156160549-bd2df891-bd02-494b-9740-d4f50a693b66.png)

![image](https://user-images.githubusercontent.com/43885625/156160728-78fc0021-95b7-4c54-b3a9-ee9ae3e5f57c.png)

![image](https://user-images.githubusercontent.com/43885625/156160813-a165603f-d265-4fdf-9d92-aabb61bba83c.png)

**Comment by xinliupitt** [comment]
I am having the same bug when running the Colab in this link prediction example

https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/gcn-link-prediction.html

**Comment by xinliupitt** [comment]
I found one solution. In `model.compile` part, the metric should be replaced using this line:
`metrics=[keras.metrics.BinaryAccuracy()]`

You may refer to the doc here:
https://keras.io/api/metrics/accuracy_metrics/

It is also weird that `keras.metrics.Accuracy()` still returns accuracy 0.0 when I tried it.

@amrlnic
